tests/SpecTests/ClientSideEncryptionSpecTest.php

```diff
@@ -0,0 +1,757 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests;
+
+use Closure;
+use Exception;
+use MongoDB\BSON\Binary;
+use MongoDB\BSON\Int64;
+use MongoDB\Client;
+use MongoDB\Driver\ClientEncryption;
+use MongoDB\Driver\Exception\BulkWriteException;
+use MongoDB\Driver\Exception\ConnectionException;
+use MongoDB\Driver\Exception\ConnectionTimeoutException;
+use MongoDB\Driver\Exception\RuntimeException;
+use MongoDB\Driver\WriteConcern;
+use MongoDB\Operation\CreateCollection;
+use MongoDB\Tests\CommandObserver;
+use PHPUnit\Framework\SkippedTestError;
+use stdClass;
+use UnexpectedValueException;
+use function base64_decode;
+use function basename;
+use function file_get_contents;
+use function getenv;
+use function glob;
+use function iterator_to_array;
+use function json_decode;
+use function sprintf;
+use function str_repeat;
+use function strlen;
+use function unserialize;
+
+/**
+ * Client-side encryption spec tests.
+ *
+ * @see https://github.com/mongodb/specifications/tree/master/source/client-side-encryption
+ */
+class ClientSideEncryptionSpecTest extends FunctionalTestCase
+{
+    const LOCAL_MASTERKEY = 'Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk';
+
+    /**
+     * Assert that the expected and actual command documents match.
+     *
+     * Note: this method may modify the $expected object.
+     *
+     * @param stdClass $expected Expected command document
+     * @param stdClass $actual   Actual command document
+     */
+    public static function assertCommandMatches(stdClass $expected, stdClass $actual)
+    {
+        static::assertDocumentsMatch($expected, $actual);
+    }
+
+    public function assertSameDocument($expectedDocument, $actualDocument)
```

Comment:
`assertOutcomeCollectionData` calls `assertSameDocument`, which won't work for the FLE test cases as the expected document contains `$$type` placeholders to only check for binary data, as it can't check against the encrypted value when a non-deterministic algorithm is used. I opted to only change this behaviour for this test instead of changing `assertOutcomeCollectionData` for all spec tests.

Reply:
If this is mainly for the outcome collection assertion, I'd prefer overriding `assertOutcomeCollectionData` in ClientSideEncryptionSpecTest.php (or making the base implementation configurable). That will ensure that any other assertions (e.g. ResultsExpectation) continue to use the correct logic.

Reply:
Good point. I made this configurable in `assertOutcomeCollectionData` and added the new assertion type in our `ResultExpectation` class.